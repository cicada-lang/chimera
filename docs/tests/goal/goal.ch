clause Exp(var(name)) -- { String(name) }
clause Exp(fn(name, ret)) -- { String(name) Exp(ret) }
clause Exp(ap(target, arg)) -- { Exp(target) Exp(arg) }

print Exp
print Exp(quote var("x"))

// TypeConstraint

print Number
print Number(1)
