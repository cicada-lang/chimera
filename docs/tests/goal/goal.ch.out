$Relation(Exp, 1)
$Goal(Exp(var("x")))
$TypeConstraint(Number)
$Goal(Number(1))
